src/algebra/category/CommRing/adjunctions.lean

```diff
@@ -28,8 +28,10 @@ polynomials with variables `x : X`.
 -/
 def free : Type u ⥤ CommRing :=
 { obj := λ α, of (mv_polynomial α ℤ),
-  map := λ X Y f, ((rename f : mv_polynomial X ℤ →ₐ[ℤ] mv_polynomial Y ℤ) :
-                              (mv_polynomial X ℤ →+*   mv_polynomial Y ℤ)),
+  map := λ X Y f,
+    -- otherwise this finds `algebra_int`, which is not defeq to the one used by `mv_polynomial`.
```

Comment:
Ouch!

Reply:
This probably will go away if `rename` is generalized to be `mv_polynomial X R →ₐ[S] mv_polynomial Y R`; or if we add a vestigual `[algebra R R]` argument to `rename`. The former opens a can of worms I don't want to touch in this PR, but I would be ok with doing the latter.

Reply:
I guess we can try out the latter, yes.

Reply:
Given that this is the only place I need the hack, I'm inclined to just leave this as is. CI passed on the previous run.